content/posts/yt_idv_spin.md

```diff
@@ -0,0 +1,56 @@
+---
+title: My semester working on yt-idv through NCSA SPIN
+date: 2022-01-27T11:32:15-06:00
+lastmod: 2022-01-27T11:32:15-06:00
+author: Aidan Sochowski
+authorlink: https://github.com/sochowski/
+cover: /img/yt_idv_spin/idvcover.png
+categories:
+  - development
+tags:
+  - volume-rendering
+  - surfaces
+  - new-feature
+  - fun
+draft: false
+---
+
+A brief overview of my contributions to yt-idv this past semester.
+
+<!--more-->
+
+# My semester working on yt-idv 
+This past semester I worked on adding a few new features to [yt-idv](https://github.com/yt-project/yt_idv) through NCSA's [Student Pushing INnovation (SPIN)](http://spin.ncsa.illinois.edu/) internship program. Since I just wrapped up my semester, here's a quick summary of the largest changes I've been working on.
+
+## Isocontours
+By far the largest feature I was working on this semester was adding isocontours to yt-idv. Isocontours serve as a method to view any level set of your data by simply setting some value and tolerance
+
+We compute these isocontour layers in a fairly simple way. Currently, all of the fragment shaders in yt-idv utilize some form of ray-tracing to determine the colormap output on the screen. Our isocontour shader does the exact same thing, except it searches for the first vertex in any defined isocontour layer (within the range of some set tolerance), and renders that one on the screen. The user can define up to 32 isocontour layers (essentially an unlimited amount) by clicking "Add Layer" and can change the tolerance using the slider when the isocontour shader is selected.
+
+Here are few examples of isocontours in yt-idv in action:
+
+![](/img/yt_idv_spin/blogpost1.gif) 
+
+*A simple single layer isocontour: this gif shows the process of adding a single isocontour layer at a density value of 10^-24, using one of yt's sample datasets, IsolatedGalaxy.*
+
+![](/img/yt_idv_spin/blogpost2.gif) 
+
+*Creating isocontours with multiple layers: this gif shows the process of adding and deleting multiple isocontour layers, demonstrated on the ActiveParticleTwoSphere sample dataset.*
+
+## Spherical Rendering
+The other main feature I worked on this semester was spherical rendering in yt-idv. Many datasets are natively stored as (r, theta, phi) spherical coordinates rather than as (x, y, z) cartesian coordinates, so it would make sense for yt-idv to support these datasets, letting the user avoid the hassle of converting these datasets to utilize a cartesian coordinate system instead.
```

Comment:
YES ! I'm really excited about this feature (albeit incomplete right now) !

Reply:
YES! Me too! I think it's not too far off. Aidan's initial work definitely moved the timeline up **significantly**